README.md

```diff
@@ -0,0 +1,142 @@
+# Pactum
+## The HTTP-API specification sketchbook for pythonistas
+
+[![Circle CI](https://circleci.com/gh/olist/pactum.svg?style=svg)](https://circleci.com/gh/olist/pactum)
+
+```shell
+pip install pactum
+```
+
+With `Pactum` you can **specify** HTTP-APIs using pure python.
+
+Pactum is easy to use, easy to extend and easy to contribute:
+
+### Easy to use
+
+The only requirements to start writing an API specification with `pactum`
+is `pactum` package itself and some knowledge of python.
+
+```python
+import pactum
+
+class MyAPI(API):
+    name = 'My API'
+    versions = [...]
+```
+
+### Easy to extend
+
+Using the [visitor pattern](http://wiki.c2.com/?VisitorPattern) you can create
+exporters and extensions for any format or service you want.
+
+Take a look at [pactum/exporters/openapi.py](https://github.com/olist/pactum/blob/master/pactum/exporters/openapi.py).
+
+### Architecture
+
+Always keep [this diagram](https://github.com/olist/pactum/wiki/Architecture-Diagram) in mind when defining your APIs.
+
+
+### Tutorial
+
+Create a file called specs.py and start defining your API.
+
+You can define a `Resource` object for your API.
+
+```python
+from pactum import Action, API, Resource, Response, Version
+from pactum import fields, verbs
+
+class OrderResource(Resource):
+    fields = [
+        fields.IntegerField(name='code', required=True),
+        fields.TimestampField(name='created_at'),
+        fields.StringField(name='item')
+    ]
+resource = OrderResource()
+
+error_resource = Resource(
+    name = 'ErrorResource'
+    fields = [fields.StringField(name='error', required=False)]
+)
+```
+You can define any element of your specification by calling it directly as in
+`error_resource` or by class definition as in `MyResource` and then calling it.
+
+
+List resources are definitions of lists of the same resource.
+```python
+list_order_resource = ListResource(resource=resource)
+```
+
+You can define `Response` objects with `status`, `description`(optional)  a
+`header`(optional) and a `Resource`/`ListResource` object as `body` (optionally)...
+
+```python
+ok_response = Response(
+    status=200, description='Here is your orders list.', body=list_resource
```

Comment:
IMHO we have a bit of unnecessary code to define a `Response`, from my understanding the process is:

1. defining a resource class
2. instantiating a resource
3. instantiating a list resource
4. creating a response

IMHO for simple cases we could skip steps 2 and 3 by allowing something like:

```python
class Order(Resource):
    ...


ok_response = Response(status=200, description='whatevs', body=Order.list)
```

I'm not sure `Order.list` is the best way of doing this, but the resulting code would be easier to write. Also, for more complex cases it might be necessary to perform steps 2 and 3, so maybe both ways can work.

Reply:
There is one thing in Pactum that is different of what we do when implementing an API. Pactum relies on the premisse that a list of resources is **one** resource with an specialize use case. It's clear now that we've to improve the explanation of this subject at documentation.

Reply:
Ok, I'll wait for the improved docs to discuss this further.